README.md

```diff
@@ -22,10 +22,125 @@ The lack of a standardized, cryptographic verification process for exchange solv
 
 ## Specification
 
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
 ### Solidity Interface
 
 ```solidity
-// [Sample Interface showing methods for proof of solvency]
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.18;
+
+/**
+    @title Cryptographically Auditable Exchange standard
+    @dev See https://eips.ethereum.org/EIPS/eip-xxxx
+ */
+interface ICryptographicallyAuditable {
+    /**
+     * @dev Struct representing an address ownership proof submitted by the CEX
+     * @param cexAddress MUST be the address owned by the CEX (submitted as a string, as it can be a non-EVM address)
+     * @param chain MUST be the name of the chain name where the address belongs (e.g., ETH, BTC)
+     * @param signature MUST be the signature of the `message` signed by the address public key
+     * @param message MUST be the message signed by the address public key
+     */
+    struct AddressOwnershipProof {
+        string cexAddress;
+        string chain;
+        bytes signature;
+        bytes message;
+    }
+
+    /**
+     * @dev Struct representing an asset owned by the CEX
+     * @param assetName MUST be a unique name of the asset
+     * @param chain MUST be the name of the chain where the asset lives (e.g., ETH, BTC). The name MUST correspond to the name used in the `AddressOwnershipProof` struct
+     * @param amount MUST be the total amount of the asset that the CEX holds on a given chain
+     */
+    struct Asset {
+        string assetName;
+        string chain;
+        uint256 amount;
+    }
+
+    /**
```

Comment:
This struct is defined here but never used anywhere else in the code. 

Another idea would be to create two different structs. One for `ZKProofOfSolvency` and one `ZKProofOfInclusion` that specify the specific `publicInputs` for each proof